tmt/steps/prepare/install.py

```diff
@@ -241,15 +254,35 @@ def go(self, guest):
                 yum_check = " || true" if skip else f" && {check}"
             else:
                 yum_check = ""
-            # Check and install
-            guest.execute(
-                f"{check} || {command} install {options} "
-                f"{packages}{yum_check}")
+            if 'rpm-ostree' not in command:
+                # Check and install
+                guest.execute(
+                    f"{check} || {command} install {options} "
+                    f"{packages}{yum_check}")
+            else:
+                for package in repo_packages:
+                    check = f'rpm -q --whatprovides {package}'
+                    if skip:
+                        try:
+                            guest.execute(f"{check}")
```

Comment:
So if `--missing skip` is used we only check whether the package are already installed and do not even try to install them? I'd expect to run `rpm-ostree install {package}` and handle the exception if the command is not successful in appropriate way based on the value of the `missing` option. Is there a reason for not trying even once?

Reply:
Thanks @psss !!
My thoughts on this were two fold; first, is that the first check in the process flow here is rpm -q --whatprovides, where if there is no match the exception is raised. My thoughts were along the lines of the question "is there any chance of a match after this check command?" I thought that the answer is logically no, however I could stand to be corrected. This leads to the second reason why I implemented this, time. Because the process of even searching for a package can be time consuming, why continue when the check was made and no match was found. Even for a package where there is no match, rpm-ostree will checkout the commit tree, update the repos and perform additional processes before actually searching the repo data for a matching package. 

- [ ] Additionally to this, certainly I agree, I should update the plugin docs. Will get to this shortly, should I use a new PR?

 Centos Stream 8 and 9 have public generic qcow disks available, however the limitation is rpm-ostree deployments, perhaps we could directly integrate the [AutoSD reference qcow](https://autosd.sig.centos.org/AutoSD-9/latest/sample-images/)? This is the only public available ostree deployment I know of which is already built.

Reply:
> Because the process of even searching for a package can be time consuming, why continue when the check was made and no match was found.

The `rpm -q` check only verifies if the package is already installed on the box locally. It does not check whether it is available in the repositories. So even if it fails the package can still be installable.

Reply:
Yeah, that would make sense.